plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/IModelDiscoverer.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.ai;
+
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+
+import java.util.List;
+
+/**
+ * Engine supports model discovery
+ * @param <MODEL> model class
+ */
+public interface IModelDiscoverer<MODEL extends IModel> {
```

Comment:
Please specify nullability annotations

Reply:
Naming as well

Reply:
AIEngineModelDoscoverer

Reply:
Not part of engine directly, part of the client

Reply:
It could be used in other places as well. But if you claim it to be an implementation-specific API, perhaps you should move it to the corresponding place instead of exposing it in the public API?